nixos/modules/services/matrix/synapse.nix

```diff
@@ -1022,15 +1096,26 @@ in {
           This is done by default unless you manually configure either of those settings.
         '';
       }
-    ];
+    ]
+    ++ (map (listener: {
+      assertion = listener.path == null || (listener.path != null && (listener.bind_addresses == null && listener.port == null && listener.tls == null));
```

Comment:
Isn't the `path != null` check redundant here?

Reply:
Yes, they are. I reordered the conditions a bit and apparently failed to clean them up.